enhancements/proposals/2549-scylladbcluster-backup-repair/README.md

```diff
@@ -0,0 +1,447 @@
+	// Refer to repair documentation for details.
+	// +optional
+	Intensity *int64 `json:"intensity,omitempty"`
+
+	// parallel specifies the maximum number of ScyllaDB repair jobs that can run at the same time (on different token ranges and replicas).
+	// Each node can take part in at most one repair at any given moment. By default, the maximum possible parallelism is used.
+	// The maximal effective parallelism depends on keyspace replication strategy and cluster topology.
+	// When set to a value greater than the maximum supported by the node, parallel is capped at the maximum supported value.
+	// Refer to repair documentation for details.
+	// +optional
+	Parallel *int64 `json:"parallel,omitempty"`
+
+	// smallTableThreshold enables small table optimization for tables of size lower than the given threshold.
+	// +optional
+	SmallTableThreshold *resource.Quantity `json:"smallTableThreshold,omitempty"`
+}
+
+type LocalScyllaDBReference struct {
+    // kind is the type of resource being referenced.
+    Kind string `json:"kind"`
+    // name is the name of resource being referenced.
+    Name string `json:"name"`
+}
+
+type ScyllaDBManagerTaskSpec struct {
+    // scyllaDBCluster is a typed reference to the target cluster in the same namespace.
+    // Supported kinds are ScyllaCluster (scylla.scylladb.com/v1) and ScyllaDBCluster (scylla.scylladb.com/v1alpha1).
+    ScyllaDBCluster LocalScyllaDBReference `json:"scyllaDBCluster"`
+    
+    // type specifies the type of the task.
+    Type ScyllaDBManagerTaskType `json:"type"`
+    
+    // backup specifies the options for a backup task.
+    // +optional
+    Backup *ScyllaDBManagerBackupTaskOptions `json:"backup,omitempty"`
+    
+    // repair specifies the options for a repair task.
+    // +optional
+    Repair *ScyllaDBManagerRepairTaskOptions `json:"repair,omitempty"`
+}
+
+type ScyllaDBManagerTaskStatus struct {
+	// observedGeneration is the most recent generation observed for this ScyllaDBManagerTask. It corresponds to the
+	// ScyllaDBManagerTask's generation, which is updated on mutation by the API Server.
+	// +optional
+	ObservedGeneration *int64 `json:"observedGeneration,omitempty"`
+
+	// conditions hold conditions describing ScyllaDBManagerTask state.
+	// +optional
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
+
+	// taskID reflects the internal identification number of the task in ScyllaDB Manager state.
+	// +optional
+	TaskID *string `json:"taskID,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+// +kubebuilder:storageversion
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:printcolumn:name="AVAILABLE",type=string,JSONPath=".status.conditions[?(@.type=='Available')].status"
+// +kubebuilder:printcolumn:name="PROGRESSING",type=string,JSONPath=".status.conditions[?(@.type=='Progressing')].status"
```

Comment:
Same here, once generation == observedGeneration task can either be scheduled - and have non-nil TaskID - or be degraded.

Reply:
done